src/algorithms/meta/FilterByAssociations.h

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Simon Gardner
+
+#pragma once
+
+#include <spdlog/spdlog.h>
+#include <algorithms/algorithm.h>
+#include <string>
+#include <string_view>
+
+#include "services/log/Log_service.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+   template<class T, class TA>
```

Comment:
Can we improve code reader's experience by making a rename?
```suggestion
   template<typename ObjectT, typename AssociationObjectT>
```